.github/workflows/compile-test.yaml

```diff
@@ -14,11 +14,28 @@ jobs:
 
     - name: Check out code into the Go module directory
       uses: actions/checkout@v1
- 
+
     - name: test
       run: |
         sudo ufw allow 2200:2300/tcp
         sudo ufw enable
         sudo ufw status verbose
         GO111MODULE=on go get sigs.k8s.io/kind@v0.7.0
-        GO111MODULE=on go test -timeout 600s -v -p 1 ./...
\ No newline at end of file
+        GO111MODULE=on go test -timeout 600s -v -p 1 ./...
+
+  code-analysis:
```

Comment:
Does this run automatically with every push?

Reply:
Actually, let's move the `gofmt-check` and `run-linter` steps under the `build-test` step above.  This would avoid doing a second pull (I think).

Reply:
The thought was to run them in parallel, since any failed task in the listed sequence of tasks causes the following tasks to not run.
I merged them into the `build-step` itself.